dts/bindings/memory-controllers/atmel,sam-smc.yaml

```diff
@@ -0,0 +1,116 @@
+# Copyright (c) 2022, Basalte bv
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Atmel Static Memory Controller (SMC).
+
+  The SMC allows to interface with static-memory mapped external devices such as
+  SRAM, PSRAM, PROM, EPROM, EEPROM, LCD Module, NOR Flash and NAND Flash.
+
+  The SMC controller can have up to 4 children defining the connected external
+  memory devices. Device Tree example:
+
+  &smc {
+    status = "okay";
+    pinctrl-0 = <&smc_default>;
+    pinctrl-names = "default";
+
+    is66wv51216dbll@0 {
+      reg = <0>;
+
+      atmel,smc-write-mode = "nwe";
+      atmel,smc-read-mode = "nrd";
+      atmel,smc-setup-timing = <1 1 1 1>;
+      atmel,smc-pulse-timing = <6 6 6 6>;
+      atmel,smc-cycle-timing = <7 7>;
+    };
+  };
+
+  Finally, in order to make the memory available you will need to define new
+  memory device/s in DeviceTree:
+
+  sram1: sram@60000000 {
+      compatible = "zephyr,memory-region", "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_K(512)>;
+      zephyr,memory-region = "SRAM1";
+  };
+
+compatible: "atmel,sam-smc"
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#address-cells":
+    required: true
+    const: 1
+
+  "#size-cells":
+    required: true
+    const: 0
+
+  peripheral-id:
+    type: int
+    description: peripheral ID
+    required: true
+
+child-binding:
+  description: |
+    Child device nodes are representing devices connected to the EBI/SMC bus.
+
+  properties:
+    reg:
+      type: int
+      required: true
+      description: |
+        The device's SMC CS number.
+        Valid range: 0 - 3
+
+    atmel,smc-write-mode:
+      type: string
+      required: false
+      default: "nwe"
+      description: |
+        Selects the mode that indicates which signal is used for the write
+        operation. Reset and default value is nwe.
+      enum:
+        - "ncs"
+        - "nwe"
+
+    atmel,smc-read-mode:
+      type: string
+      required: false
+      default: "nrd"
+      description: |
+        Selects the mode that indicates which signal is used for the read
+        operation. Reset and default value is nrd.
+      enum:
+        - "ncs"
+        - "nrd"
+
+    atmel,smc-setup-timing:
+      type: array
+      required: false
+      default: [1, 1, 1, 1]
+      description: |
+        SMC setup timing configuration. [NWE, NCS_WR, NRD, NCS_RD]
+        Refer to the datasheet for more information.
+
+    atmel,smc-pulse-timing:
+      type: array
+      required: false
+      default: [1, 1, 1, 1]
+      description: |
+        SMC pulse timing configuration. [NWE, NCS_WR, NRD, NCS_RD]
+        Refer to the datasheet for more information.
+
+    atmel,smc-cycle-timing:
+      type: array
+      required: false
+      default: [3, 3]
+      description: |
+        SMC cycle timing configuration. [NWE, NRD]
+        Refer to the datasheet for more information.
```

Comment:
All of these defaults must be justified in the descriptions.  https://docs.zephyrproject.org/latest/build/dts/bindings.html#rules-for-default-values

Reply:
@mbolivar-nordic I've updated the descriptions with information from the datasheet